lib/web_ui/lib/src/engine/window.dart

```diff
@@ -72,11 +73,32 @@ base class EngineFlutterView implements ui.FlutterView {
 
   final ViewConfiguration _viewConfiguration = const ViewConfiguration();
 
+  /// Whether this [EngineFlutterView] has been disposed or not.
+  bool isDisposed = false;
+
+  /// Disposes of the [EngineFlutterView] instance and undoes all of its DOM
+  /// tree and any event listeners.
+  @mustCallSuper
+  void dispose() {
+    isDisposed = true;
+    platformDispatcher.unregisterView(this);
```

Comment:
Right now the view constructor is automatically doing `register`, and this needs to be here (but we said that registration wouldn't happen automatically upon creation, so this will go in that case, right?)

Reply:
Yep. `dispose` should undo anything that this class did. If the view is registering itself, then it should also unregister itself (in `dispose`). If someone else is registering the view, then that someone should handle the unregistering too.